### PR TITLE
Adds pipeline prompt

### PR DESCRIPTION
**Pull request by jacoblee93**
Adds a new type of prompt that can use intermediate prompts in the final output value.

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
conceptually lgtm, would be helpful to have example maybe?

**Comment by jacoblee93** [comment]
> conceptually lgtm, would be helpful to have example maybe?

Yes, will add docs before merging.

File: langchain/src/prompts/pipeline.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import { InputValues, PartialValues } from "../schema/index.js";
+import { BasePromptTemplate, BasePromptTemplateInput } from "./base.js";
+import { ChatPromptTemplate } from "./chat.js";
+import { SerializedBasePromptTemplate } from "./serde.js";
+
+export type PipelinePromptParams<T extends BasePromptTemplate> = {
```
Comment (jacoblee93) [inline langchain/src/prompts/pipeline.ts:6]: TODO: Change `T` to a more descriptive name

**Comment by jacoblee93** [comment]
> > conceptually lgtm, would be helpful to have example maybe?
> 
> Yes, will add docs before merging.

Done!

File: langchain/src/prompts/pipeline.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      .map((pipelinePrompt) =>
+        pipelinePrompt.prompt.inputVariables.filter(
+          (inputValue) => !intermediateValues.includes(inputValue)
+        )
+      )
+      .flat();
+    return [...new Set(inputValues)];
+  }
+
+  protected static extractRequiredInputValues(
```
Comment (nfcampos) [inline langchain/src/prompts/pipeline.ts:48]: Why protected on a static field?

Comment (jacoblee93) [reply langchain/src/prompts/pipeline.ts:48]: Just as documentation - that it's not intended to be part of the public interface

**Pull request merged**